### PR TITLE
[release-4.11] OCPBUGS-668: Fix bad DNS endpoint selection

### PR DESCRIPTION
**Pull request by martinkennelly**
Previous to this change, the OCP specific hack in
openshift/kubernetes that this component vendors in
is broken due to incorrect rebase.
If a local DNS endpoint exists, it was not selected.

Signed-off-by: Martin Kennelly <mkennell@redhat.com>

### CODE REVIEW

**Comment by martinkennelly** [comment]
/retest 

Test failure in e2e-aws-upgrade unrelated to this PR: "openshift cluster install failed with infrastructure setup"

**Comment by martinkennelly** [comment]
I tested this PR with a cluster manually to verify it selects DNS endpoints locally consistently. I also verified it falls back to an endpoint on a different node when a local endpoint is not available.

**Comment by danwinship** [comment]
/label backport-risk-assessed

o/k PR is merged. You'll have to figure out how to appease the bot... (Easiest approach may be to create a dummy 4.11 bug and just close it, noting that the bug only existed in 4.10.)

**Comment by martinkennelly** [comment]
@anuragthehatter Can you review to add `/label cherry-pick-approved` ?

**Comment by danwinship** [comment]
Oh, so I was thinking this was a 4.10 bug because of the o/k branch name, but I forgot that 4.11 ended up shipping with the 4.10 branch still... anyway, we'll need to backport this fix to 4.10 openshift-sdn too. (Can use the same o/k branch.)

File: go.mod [hunk 1]
```diff
@@ -71,33 +71,33 @@ replace (
...
+	k8s.io/cri-api => github.com/openshift/kubernetes/staging/src/k8s.io/cri-api v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/csi-translation-lib => github.com/openshift/kubernetes/staging/src/k8s.io/csi-translation-lib v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kube-aggregator => github.com/openshift/kubernetes/staging/src/k8s.io/kube-aggregator v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kube-controller-manager => github.com/openshift/kubernetes/staging/src/k8s.io/kube-controller-manager v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kube-proxy => github.com/openshift/kubernetes/staging/src/k8s.io/kube-proxy v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kube-scheduler => github.com/openshift/kubernetes/staging/src/k8s.io/kube-scheduler v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kubectl => github.com/openshift/kubernetes/staging/src/k8s.io/kubectl v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kubelet => github.com/openshift/kubernetes/staging/src/k8s.io/kubelet v0.0.0-20220830115746-9443ebdc888a
+	k8s.io/kubernetes => github.com/openshift/kubernetes v0.0.0-20220830115746-9443ebdc888a
```
Comment (danwinship) [inline go.mod:94]: I don't know why go.mod is confused about the version number, but the git hash is correct...

**Comment by martinkennelly** [comment]
/retest 

Infra setup error

**Pull request merged**